x-pack/plugins/security/server/authentication/providers/saml.ts

```diff
@@ -269,10 +282,9 @@ export class SAMLAuthenticationProvider extends BaseAuthenticationProvider {
       });
 
       this.logger.debug('Login has been performed with SAML response.');
-      return AuthenticationResult.redirectTo(
-        stateRedirectURL || `${this.options.basePath.get(request)}/`,
-        { state: { accessToken, refreshToken } }
-      );
+      return AuthenticationResult.redirectTo(nextURL || `${this.options.basePath.get(request)}/`, {
```

Comment:
__note:__ we should probably validate if it's a relative path at least. What do you think?